### PR TITLE
Added all tests described in Writing Tests - Part 1

### PR DESCRIPTION
**Pull request by aparnaroy**
Tested the database, home page, timeline page, and edge cases on the timeline page.

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -9,14 +9,17 @@
...
+    print("Running in test mode")
+    mydb = SqliteDatabase('file:memory?mode=memory&cache=shared', uri=True)
+else:
+    mydb = MySQLDatabase(
+        os.getenv('MYSQL_DATABASE'),
+        user=os.getenv('MYSQL_USER'),
+        password=os.getenv('MYSQL_PASSWORD'),
+        host=os.getenv('MYSQL_HOST'),
+        port=3306
+    )
```
Comment (pablo-ramm) [inline app/__init__.py:22]: Good idea to have a in memory db when in testing mode

File: app/__init__.py [hunk 2]
```diff
@@ -111,10 +114,23 @@ def experience_page():
...
+        email = request.form['email']
+    if not request.form['content'] or request.form['content'] == "":
+        return "Invalid content", 400
+    else:
+        content = request.form['content']
+    
+    valid_email = "@" in email and "." in email
+    if valid_email == False:
+        return "Invalid email", 400
+    timeline_post =TimelinePost.create(name=name, email=email, content=content)
```
Comment (pablo-ramm) [inline app/__init__.py:133]: Excellent, checking for all parameters in the request to the API

File: tests/test_app.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        response = self.client.post("/api/timeline_post", data={"name": "John Doe", "email": "john@example.com", "content": ""})
+        assert response.status_code == 400
+        html = response.get_data(as_text=True)
+        assert "Invalid content" in html
+
+        # POST request with malformed email
+        response = self.client.post("/api/timeline_post", data={"name": "John Doe", "email": "not-an-email", "content": "Hello world, I'm John!"})
+        assert response.status_code == 400
+        html = response.get_data(as_text=True)
+        assert "Invalid email" in html
```
Comment (pablo-ramm) [inline tests/test_app.py:81]: The tests look solid and test a lot of functionalities of the site

File: tests/test_db.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        second_post = TimelinePost.create(name='Jane Doe', email='jane@example.com', content='Hello world, I\'m Jane!')
+        assert second_post.id == 2
+
+        # TODO: Get timeline posts and assert that they are correct
+        posts = [
+            model_to_dict(p)
+            for p in TimelinePost.select().order_by(TimelinePost.created_at.desc())
+        ]
+        assert posts[0] == model_to_dict(second_post)
+        assert posts[1] == model_to_dict(first_post)
```
Comment (pablo-ramm) [inline tests/test_db.py:32]: This file is very important because it ensures the db is working as expected

**Review by pablo-ramm — APPROVED** [review]
The PR was solid and easy to followe. :)
